toolchain/driver/driver.cpp

```diff
@@ -248,8 +266,16 @@ auto Driver::RunDumpSubcommand(DiagnosticConsumer& consumer,
 
   if (dump_mode == DumpMode::Assembly) {
     consumer.Flush();
-    has_errors |= !Carbon::PrintAssemblyFromModule(
-        *module, target_triple, error_stream_, output_stream_);
+    CodeGen codegen =
+        CodeGen(*module, target_triple, error_stream_, output_stream_);
+    has_errors |= !codegen.PrintAssembly();
+    return !has_errors;
+  }
+
+  if (dump_mode == DumpMode::ObjectCode) {
+    CodeGen codegen =
+        CodeGen(*module, target_triple, error_stream_, output_stream_);
+    has_errors |= !codegen.GenerateObjectCode(output_file);
```

Comment:
Our behavior would be more uniform if we set `output_stream_` to be the output file here, and don't pass a separate output file to `GenerateObjectCode`.

Reply:
But if we don't provide an output file to write the object code into how do we generate the binary?

Reply:
I'm thinking of something like this:
```
    std::error_code ec;
    llvm::raw_fd_ostream dest(output_file, ec, llvm::sys::fs::OF_None);
    if (ec) {
      error_stream_ << "Error: Could not open file: " << ec.message() << "\n";
      return false;
    }
    CodeGen codegen(*module, target_triple, error_stream_, dest);
    has_errors |= !codegen.GenerateObjectCode();
```